src/pg/extensions/citext.cr

```diff
@@ -0,0 +1,62 @@
+require "../../pg"
+
+module PG
+  module Extension
+    # This extension adds support for decoding the Postgres `citext` type.
+    class CIText
+      include Extension
+
+      def load(connection)
+        oid = connection.query_one "SELECT oid FROM pg_type WHERE typname = 'citext'", as: UInt32
+        connection.register_decoder Decoder.new([oid.to_i])
+      end
+
+      struct Decoder
+        include Decoders::Decoder
+
+        getter oids : Array(Int32)
+
+        def initialize(@oids : Array(Int32))
+        end
+
+        def decode(io, bytesize, oid)
+          PG::CIText.new(Decoders::StringDecoder.new.decode(io, bytesize, oid))
+        end
+
+        def type
+          PG::CIText
+        end
+      end
+    end
+
+    Connection.register_extension CIText.new
+  end
+
+  struct CIText
+    def initialize(text : String)
+      @text = text.downcase
```

Comment:
This `downcase` might be leftover from an earlier version of this code and may not be necessary. If we don’t need to downcase, that’ll save us a heap allocation.